File: gestao-qualidade-software/2026-2/atividades/1009/consulta.py
# Fatia didatica — consulta de escolas. Nao e o sitio BDGETEC. Sem HTTP.
# NAO "consertar" buscar/contar/listar/ativas hoje.
# Execute um comando por vez; consulte aluno.md.

ESCOLAS_NRA1 = ["Etec Sao Paulo", "Etec Zona Leste", "Etec de Artes"]
PERIODO_CASO = "2026/2"


def buscar(regional, periodo):
    if regional == "NRA1" and periodo == "2026/1":
        return list(ESCOLAS_NRA1)
    return []


def contar(regional, periodo):
    buscar(regional, periodo)
    return 3


def listar(regional):
    escolas = list(ESCOLAS_NRA1) if regional == "NRA1" else []
    linhas = []
    i = 0
    while i <= len(escolas):
        linhas.append(str(i + 1) + " - " + escolas[i])
        i += 1
    return linhas


def ativas(regional, periodo):
    return list(ESCOLAS_NRA1)


def oraculo_buscar():
    # esperado: buscar("NRA1", "2026/2") devolve escolas
    obtido = buscar("NRA1", PERIODO_CASO)
    ok = len(obtido) > 0
    print("PASS" if ok else "FAIL")
    return 0 if ok else 1


def oraculo_contar():
    # esperado independente: contar("XXX", "2026/2") == 0
    # TODO: implemente PASS/FAIL + return 0/1. Nao altere as funcoes.
    print("TODO")
    return 2


def mostrar_regional():
    casos = [("NRA1", "2026/1"), ("nra1", "2026/1"), (" NRA1", "2026/1")]
    for regional, periodo in casos:
        print(repr(regional), "+", periodo, "->", len(buscar(regional, periodo)))


def mostrar_ativas():
    print(ativas("XXX", PERIODO_CASO))


def main(argv):
    cmd = argv[1] if len(argv) > 1 else "buscar"
    if cmd == "buscar":
        raise SystemExit(oraculo_buscar())
    if cmd == "contar":
        print(contar("NRA1", PERIODO_CASO))
        raise SystemExit(0)
    if cmd == "listar":
        print("\n".join(listar("NRA1")))
        raise SystemExit(0)
    if cmd == "regional":
        mostrar_regional()
        raise SystemExit(0)
    if cmd == "ativas":
        mostrar_ativas()
        raise SystemExit(0)
    if cmd == "oraculo-contar":
        raise SystemExit(oraculo_contar())
    print("uso: python3 consulta.py buscar|contar|listar|regional|ativas|oraculo-contar")
    raise SystemExit(2)


if __name__ == "__main__":
    import sys
    main(sys.argv)

File: gestao-qualidade-software/2026-2/atividades/1009/consulta.js
// Fatia didatica — consulta de escolas. Nao e o sitio BDGETEC. Sem HTTP.
// NAO "consertar" buscar/contar/listar/ativas hoje.
// Execute um comando por vez; consulte aluno.md.

const ESCOLAS_NRA1 = ["Etec Sao Paulo", "Etec Zona Leste", "Etec de Artes"];
const PERIODO_CASO = "2026/2";

function buscar(regional, periodo) {
  if (regional === "NRA1" && periodo === "2026/1") {
    return ESCOLAS_NRA1.slice();
  }
  return [];
}

function contar(regional, periodo) {
  buscar(regional, periodo);
  return 3;
}

function listar(regional) {
  const escolas = regional === "NRA1" ? ESCOLAS_NRA1.slice() : [];
  const linhas = [];
  for (let i = 0; i <= escolas.length; i++) {
    if (escolas[i] === undefined) {
      throw new Error("indice " + i + " fora da lista");
    }
    linhas.push(i + 1 + " - " + escolas[i]);
  }
  return linhas;
}

function ativas(regional, periodo) {
  return ESCOLAS_NRA1.slice();
}

function oraculoBuscar() {
  const obtido = buscar("NRA1", PERIODO_CASO);
  const ok = obtido.length > 0;
  console.log(ok ? "PASS" : "FAIL");
  return ok ? 0 : 1;
}

function oraculoContar() {
  // esperado independente: contar("XXX", "2026/2") === 0
  // TODO: implemente PASS/FAIL + return 0/1. Nao altere as funcoes.
  console.log("TODO");
  return 2;
}

function mostrarRegional() {
  const casos = [
    ["NRA1", "2026/1"],
    ["nra1", "2026/1"],
    [" NRA1", "2026/1"],
  ];
  for (const [regional, periodo] of casos) {
    console.log(JSON.stringify(regional) + " + " + periodo + " -> " + buscar(regional, periodo).length);
  }
}

const cmd = process.argv[2] || "buscar";
if (cmd === "buscar") process.exit(oraculoBuscar());
if (cmd === "contar") {
  console.log(contar("NRA1", PERIODO_CASO));
  process.exit(0);
}
if (cmd === "listar") {
  console.log(listar("NRA1").join("\n"));
  process.exit(0);
}
if (cmd === "regional") {
  mostrarRegional();
  process.exit(0);
}
if (cmd === "ativas") {
  console.log(ativas("XXX", PERIODO_CASO).join(", "));
  process.exit(0);
}
if (cmd === "oraculo-contar") process.exit(oraculoContar());
console.log("uso: node consulta.js buscar|contar|listar|regional|ativas|oraculo-contar");
process.exit(2);
